### PR TITLE
User Highlight priority should be: friend > OP > mod > etc...

### PR DESCRIPTION
**Pull request by githue**
userhighlight-priority

Priority of colors, as seen on vanilla reddit.

### CODE REVIEW

File: lib/modules/userHighlight.js [hunk 1]
```diff
@@ -186,12 +186,12 @@ modules['userHighlight'] = {
 					hoverColor: this.options.firstCommentColorHover.value
 				}
 			};
-			if (this.options.highlightOP.value) {
-				this.highlight('submitter');
-			}
 			if (this.options.highlightFriend.value) {
```
Comment (mc10) [inline lib/modules/userHighlight.js]: Shouldn't we be flipping this, doing mod, then submitter, then friend? Seems like the latest rule should trump all the rest.

File: lib/modules/userHighlight.js [hunk 2]
```diff
@@ -161,14 +161,14 @@ modules['userHighlight'] = {
 		this.options['generateHoverColors'].callback = modules['userHighlight'].generateHoverColors;
 		if ((this.isEnabled()) && (this.isMatchURL())) {
 			this.colorTable = {
```
Comment (mc10) [inline lib/modules/userHighlight.js]: I think it's best to just alphabetize the elements here.

File: lib/modules/userHighlight.js (hunk 1 above)
Comment (githue) [inline lib/modules/userHighlight.js]: In the screenshot the styles are ordered top-to-bottom in order of highest priority to lowest, it's from DevTools which re-orders selectors for convenience. If you were to look at the raw CSS it should look like:

```
.friend {}
.submitter {}
.moderator {}
.admin {}.
```

The if statements should be in that order too. Sorry for the confusion.

**Comment by andytuba** [comment]
This'll be nice to clean up.   I personally would have sorted it the other way around: admin > mod > friend > op, in order of most exceptional to most regular. What's your thinking/rationale/source for the order here?

**Comment by erikdesjardins** [comment]
@andytuba the title is in reverse order (i.e. ">" = "is superseded by"), so it's the same as your suggestion, except OP overrides Friend (a la reddit).
image

**Comment by andytuba** [comment]
Oh, okay, thanks for the clue :)

**Comment by githue** [comment]
Yeah I didn't put much thought into the title, Erik is right.

**Comment by andytuba** [comment]
@andytuba TODO changeloge

**Pull request merged**